samples/convert-upload-images-powervs/convert-upload-images-powervs

```diff
@@ -0,0 +1,742 @@
+  cp -rf "infra-$ARTIFACTS_VERSION" ./"$ARTIFACTS_DIR-1"
+  rm -rf "infra-$ARTIFACTS_VERSION"
+  mv ./"$ARTIFACTS_DIR"-1/infra-node ./"$ARTIFACTS_DIR"
+  rm -rf ./"$ARTIFACTS_DIR"-1
+}
+
+
+#-------------------------------------------------------------------------
+# Configure cos resource crn
+#-------------------------------------------------------------------------
+function get_cos_service_id {
+  log "get crn for cos instance"
+  $CLI_PATH plugin install cloud-object-storage -q -f  > /dev/null
+  COS_INSTANCE_ID=$($CLI_PATH resource service-instance "${COS_INSTANCE_NAME}" --id | grep 'crn' | awk '{print $2}')
+  #$CLI_PATH cos config crn --crn "${COS_CRN}" --force > /dev/null
+}
+
+#-------------------------------------------------------------------------
+# Destroy VM
+#-------------------------------------------------------------------------
+function destroy_vm {
+    TF='./terraform'
+    SECONDS=0
+    message="VM Destroy completed in"
+    destroy # Destroying the VM
+    time_taken "$message"
+}
+
+#-------------------------------------------------------------------------
+# Create VM
+#-------------------------------------------------------------------------
+function create_vm {
+  PWD_SAVED="$PWD"
+  SECONDS=0
+  message="VM Creation completed in"
+  apply
+  time_taken "$message"
+  cd $PWD_SAVED # Going back to the script directory
+
+}
+
+#-------------------------------------------------------------------------
+# Download and setup pvsadm tool in the local machine
+#-------------------------------------------------------------------------
+function setup_pvsadm_local {
+  log "Identifying operating system and settingup pvsadm"
+  if [[ "$OS" == "darwin" ]]; then
+    download_url "https://github.com/ppc64le-cloud/pvsadm/releases/download/${PVSADM_VERSION}/pvsadm-darwin-amd64.tar.gz"
+    tar -xzvf $IMAGE_PATH >/dev/null 2>&1
+    export PATH=./:$PATH
+    pvsadm version
+  fi
+  if [[ "$OS" == "linux" ]]; then
+    download_url "https://github.com/ppc64le-cloud/pvsadm/releases/download/${PVSADM_VERSION}/pvsadm-linux-amd64.tar.gz"
+    tar -xzvf $IMAGE_PATH >/dev/null 2>&1
+    export PATH=./:$PATH
+    pvsadm version
+  fi
+  if [[ "$OS" == "windows" ]]; then
+    download_url "https://github.com/ppc64le-cloud/pvsadm/releases/download/${PVSADM_VERSION}/pvsadm-windows-amd64.tar.gz"
+    tar -zxvf $IMAGE_PATH >/dev/null 2>&1
+    export PATH=./:$PATH
+    ./pvsadm version
+  fi
+}
+
+#-------------------------------------------------------------------------
+# Download and setup pvsadm tool in the remote CentOS instance
+#-------------------------------------------------------------------------
+function setup_pvsadm_remote {
+  # Sets up pvsadm in the VM created
+  LOG_FILE="logs/setup_pvsadm_remote_${i}_${LOGFILE}.log"
+  log "Identifying operating system and settingup pvsadm"
+  ARCH=$(ssh -o 'StrictHostKeyChecking no' -i $VAR_PRIVATE_KEY_FILE root@${BASTION_PUBLIC_IP} "arch") 2>&1
+  if [[ "$ARCH" == "ppc64le" ]]; then
```

Comment:
What happens when `ARCH=x86_64`? If x86 is not supported then why check for remote ARCH?

Reply:
yea, true, this check is not need